src/commands.ts

```diff
@@ -60,51 +62,58 @@ async function buildCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver:
         project = args.fsPath;
     }
 
+    await buildProjects(dotNetSdk, [project], commandObserver, cancelToken).then(() => {
+        commandObserver.buildInProgress = false;
+    });
+}
+
+async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
+    await buildProjects(dotNetSdk, projects.map(p => p.fsPath), commandObserver, cancelToken);
+}
+
+async function buildProjects(dotNetSdk: DotNetInfo, projects: string[], commandObserver: CommandObserver, cancelToken: vscode.CancellationToken) {
     commandObserver.clear();
+    let unsupportedProjects = await validateProjectSDK(projects, commandObserver);
+    if (unsupportedProjects.length > 0) {
+        projects = projects.filter(p => unsupportedProjects.indexOf(p) < 0);
+    }
 
-    let isValid = await validateProjectSDK([project], commandObserver);
-    if (!isValid) {
+    if (commandObserver.buildInProgress) {
+        vscode.window.showErrorMessage(Constants.existingBuildInProgressMessage);
         return;
     }
-    await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken);
-}
 
-async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
     try {
-        let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
-        commandObserver.clear();
-
-        let isValid = await validateProjectSDK(projects.map(p => p.fsPath), commandObserver);
-        if (!isValid) {
-            return;
-        }
-
+        commandObserver.buildInProgress = true;
```

Comment:
This should probably be the first thing we set after we check for already running build.